include/boost/wave/util/cpp_macromap.hpp

```diff
@@ -1233,9 +1302,21 @@ ContainerT replacement_list;
                 macro_def.macroparameters.size(), seen_newline);
 #endif
 
+        std::size_t parm_count_required = macro_def.macroparameters.size();
+#if BOOST_WAVE_SUPPORT_CPP2A
+        if (boost::wave::need_cpp2a(ctx.get_language())) {
+            // if the last parameter is ellipsis (varargs),
+            // reduce the required count by one
+            if ((parm_count_required > 0) &&
+                (T_ELLIPSIS == token_id(macro_def.macroparameters.back()))) {
+                --parm_count_required;
+            }
+        }
+#endif
```

Comment:
Would you mind adding a comment explaining that in C++20 the rules have changed for how many arguments are required if ellipses are involved? I.e. something like:
> The number of arguments must be more than (until C++20) / at least as many as (since C++20) the number of parameters (not counting ...).

Reply:
Sure, no problem. Going with:

>            // Before C++20, at least one argument must be supplied for
>            // the variable parameters. So reduce the mandatory
>            // argument count by one if the last parameter is ellipsis: